File: app-geekhome/src/main/java/eu/geekhome/rest/hardware/AdapterController.java
package eu.geekhome.rest.hardware;

import eu.geekhome.HardwareManager;
import eu.geekhome.rest.HardwareManagerHolderService;
import eu.geekhome.services.hardware.HardwareAdapterDto;

import javax.inject.Inject;
import javax.ws.rs.GET;
import javax.ws.rs.Path;
import javax.ws.rs.Produces;
import javax.ws.rs.core.MediaType;
import java.util.List;
import java.util.stream.Collectors;

@Path("hardwareadapters")
public class AdapterController {

    private final HardwareManager _hardwareManager;
    private HardwareAdapterDtoMapper _mapper;

    @Inject
    public AdapterController(
            HardwareManagerHolderService hardwareManagerHolderService,
            HardwareAdapterDtoMapper mapper) {
        _hardwareManager = hardwareManagerHolderService.getInstance();
        _mapper = mapper;
    }

    @GET
    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    public List<HardwareAdapterDto> getAdapters() {
        return _hardwareManager
                    .getFactories()
                    .stream()
                    .map(_mapper::map)
                    .collect(Collectors.toList());
    }
}

File: app-geekhome/src/main/java/eu/geekhome/rest/hardware/AdapterController.kt
package eu.geekhome.rest.hardware

import eu.geekhome.domain.hardware.HardwareManager
import eu.geekhome.domain.events.EventsSink
import javax.inject.Inject
import eu.geekhome.data.hardware.HardwareAdapterDto
import javax.ws.rs.*
import javax.ws.rs.core.MediaType

@Path("hardwareadapters")
class AdapterController @Inject constructor(
    private val hardwareManager: HardwareManager,
    private val mapper: HardwareAdapterDtoMapper,
    private val liveEvent: EventsSink
) {

    @Produces(MediaType.APPLICATION_JSON + ";charset=utf-8")
    @GET
    fun getAdapters(): List<HardwareAdapterDto> {
        return hardwareManager
            .bundles()
            .map { mapper.map(it) }
    }

    @PUT
    @Path("/{factoryId}/discover")
    @Produces(MediaType.APPLICATION_JSON + ";charset=UTF-8")
    fun discover(@PathParam("factoryId") factoryId: String) {
        hardwareManager.scheduleDiscovery(factoryId)
    }
}